ipatests/test_integration/test_sysaccounts.py

```diff
@@ -427,3 +474,672 @@ def test_system_account_modify_password(self):
+            # the sysaccount dn on the master
+            result = tasks.ldapsearch_dm(
+                self.master,
+                config_dn,
+                ["(objectclass=*)", "sysacctmanagersdns"],
+                scope='base'
+            )
+            assert f"sysacctmanagersdns: {sysaccount_dn}" in result.stdout_text
+
+            # Check that ldapmodify can remove the value
+            entry_ldif = textwrap.dedent("""
+                dn: {config_dn}
+                changetype: modify
+                delete: sysacctmanagersdns
+                sysacctmanagersdns: {sysaccount_dn}
+            """).format(config_dn=config_dn, sysaccount_dn=sysaccount_dn)
+
+            self.master.run_command(
+                [
+                    "ldapmodify",
+                    "-x",
+                    "-D", str(self.master.config.dirman_dn),
+                    "-w", self.master.config.dirman_password,
+                    "-H", f"ldap://{self.master.hostname}",
+                ],
+                stdin_text=entry_ldif
+            )
+
+            # Verify the value is removed
+            result = tasks.ldapsearch_dm(
+                self.master,
+                config_dn,
+                ["(objectclass=*)", "sysacctmanagersdns"],
+                scope='base'
+            ).stdout_text
+            assert f"sysacctmanagersdns: {sysaccount_dn}" not in result
+        finally:
+            # Clean up - ensure sysaccount is removed if it still exists
+            tasks.kinit_admin(self.master)
+            self.master.run_command(
+                ["ipa", "sysaccount-del", sysaccount_name], raiseonerr=False
+            )
+
+
+@contextlib.contextmanager
+def full_backup_restore(host, data_only=False):
+    """Context manager for full backup, uninstall, and restore
+
+    :param host: host object
+    :param data_only: if True, create data-only backup and tasks.ipa_restore
+                      if False, create full backup, uninstall, and restore
+    """
+    # Create backup
+    backup_path = tasks.get_backup_dir(host, data_only=data_only)
+
+    if data_only:
+        # For data-only backup, just yield and let caller handle restore
+        yield
+        # Restore from backup
+        tasks.ipa_restore(host, backup_path)
+    else:
+        # For full backup, uninstall and restore
+        # Uninstall IPA
+        host.run_command(['ipa-server-install', '--uninstall', '-U'])
+
+        # Restore from backup
+        dirman_password = host.config.dirman_password
+        host.run_command(
+            ['ipa-restore', backup_path],
+            stdin_text=f'{dirman_password}\nyes'
+        )
+
+        yield
+
+    # Wait for services and re-authenticate
+    tasks.wait_for_sssd_domain_status_online(host)
+    tasks.kinit_admin(host)
+
+
+class TestSystemAccountsBackupRestore(IntegrationTest):
+    """Integration tests for system accounts with backuprestore"""
+
```

Comment:
For this class of tests you can use a strategy similar to the one in the file test_backup_and_restore.py:
in the setup, prepare all the data you need (a privileged sysaccount, an unprivileged sysaccount, a disabled sysaccount etc...)
- define a restore_checker that is a context manager: loop on a list of checks, store the result, yield, loop on the list of checks and compare the results
- each check can represent an item you want the test to compare, for instance the output of ipa sysaccount-show "privilegedsysaccount"

The test will be faster as it calls ipa-backup and ipa-restore a single time.